src/interface/src/app/plan/plan-summary/saved-scenarios/saved-scenarios.component.ts

```diff
@@ -57,26 +57,27 @@ export class SavedScenariosComponent implements OnInit {
   }
 
   viewScenario(id?: string): void {
-    if (!id) {
-      id = this.scenarios.find((scenario) => scenario.selected)?.id;
-    }
-    this.router.navigate(['scenario', id], { relativeTo: this.route });
+    this.router.navigate(['config', this.highlightedId], { relativeTo: this.route });
+  }
+
+  highlightScenario(id: string): void {
+    this.highlightedId = id;
   }
 
   showDeleteButton(): boolean {
     return this.scenarios.filter((scenario) => scenario.selected).length > 0;
   }
 
   showViewButton(): boolean {
-    return this.scenarios.filter((scenario) => scenario.selected).length === 1;
+    return this.highlightedId!='-1';
   }
 
   deleteSelectedScenarios(): void {
     this.planService
       .deleteScenarios(
         this.scenarios
           .filter((scenario) => scenario.selected)
-          .map((scenario) => scenario.id)
+          .map((scenario) => String(scenario.id))
```

Comment:
For this call, I ended up keeping the service call the same, since it ultimately ended up taking highlightedId which is actually a string. 

For clarification, do you prefer having old and new versions of methods to just updating the methods to use the new types?

Reply:
Ideally replace the methods to use new types, but not sure how much of a refactor effort that is